src/ooxml/java/org/apache/poi/xwpf/usermodel/XWPFChart.java

```diff
@@ -1,123 +1,343 @@
+         * performance issue, two possible approaches are available:<br>
+         * a) Use the checksum only and take the risk that two images might have
+         * the same CRC32 sum, although they are not the same.<br>
+         * b) Use a second (or third) checksum algorithm to minimise the chance
+         * that two images have the same checksums but are not equal (e.g.
+         * CRC32, MD5 and SHA-1 checksums, additionally compare the
+         * data-byte-array lengths).
+         */
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (!(obj instanceof XWPFChart)) {
+            return false;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return getChecksum().hashCode();
+    }
+    
+    /**
+     * method to create relationship with embedded part
+     * for example writing xlsx file stream into output stream
+     * @param chartSheet
+     * @return return relation part which used to write relation in .rels file and get relation id
+     * @since POI 4.0
+     */
+	public RelationPart createRelationship(XWPFRelation chartRelation,int index) {
+		return createRelationship(chartRelation, XWPFFactory.getInstance(), index, false);
+	}
+	
+	/**
+	 * protected method which used to initialization of sheet output stream 
+	 * @param sheet
+	 * @since POI 4.0
+	 */
+	protected void addEmbeddedWorkSheet(OutputStream sheet) {
+			this.sheet=sheet;
+	}
+	
+	/**
+	 * this method is used to write workbook object in embedded part of chart
+	 * return's true in case of successfully write work book in embedded part or return's false
+	 * @param wb
+	 * @return return's true in case of successfully write work book in embedded part or return's false
+	 * @throws IOException
+	 * @since POI 4.0
+	 */
+	public boolean writeEmbeddedWorkSheet(Workbook wb) throws IOException
+	{
+		if(this.sheet!=null && wb!=null)
+		{
+			wb.write(this.sheet);
+			return true;
+		}
+		return false;
+	}
+	
+	/**
+	 * initialize in line object
+	 * @param inline
+	 * @since POI 4.0
+	 */
+	protected void setInLine(CTInline ctInline) {
+		this.ctInline=ctInline;
+	}
+	
+	/**
+	 * set chart height 
+	 * @param height
+	 * @since POI 4.0
+	 */
+	public void setChartHeight(long height)
+	{
+		ctInline.getExtent().setCy(height);
+	}
+	
+	/**
+	 * set chart width 
+	 * @param width
+	 * @since POI 4.0
+	 */
+	public void setChartWidth(long width)
+	{
+		ctInline.getExtent().setCx(width);
+	}
+	/**
+	 * get chart height 
+	 * @since POI 4.0
+	 */
+	public long getChartHeight()
+	{
+		return ctInline.getExtent().getCy();
+	}
+	
+	/**
+	 * get chart width 
+	 * @since POI 4.0
+	 */
+	public long getChartWidth()
+	{
+		return ctInline.getExtent().getCx();
+	}
+	
+	/**
+	 * set chart height and width
+	 * @param width
+	 * @param height 
+	 * @since POI 4.0
+	 */
+	public void setChartWidthHeight(long width,long height)

```

Comment:
Is this necessary? I'd suggest removing this.

Reply:
If you really have a motivation to maintain this method, at least rename it to something like `setChartBoundingBox`.